db.go

```diff
@@ -1351,7 +1359,9 @@ func (db *DB) KeySplits(prefix []byte) []string {
 		for _, mt := range memTables {
 			mtSplits(mt)
 		}
-		mtSplits(db.mt)
+		if !db.opt.ReadOnly {
+			mtSplits(db.mt)
```

Comment:
Add a `if nil return` check in `mtsplit`. It looks much cleaner.

Reply:
thanks. done.